File: typescript/resistor-color-duo/resistor-color-duo.test.ts
import { ResistorColor } from "./resistor-color-duo"

describe("Resistor Colors", () => {
  test("Brown and black", () => {
    const resistorColor = new ResistorColor(["brown", "black"])
    expect(resistorColor.value()).toEqual(10)
  })

  test("Blue and grey", () => {
    const resistorColor = new ResistorColor(["blue", "grey"])
    expect(resistorColor.value()).toEqual(68)
  })

  test("Yellow and violet", () => {
    const resistorColor = new ResistorColor(["yellow", "violet"])
    expect(resistorColor.value()).toEqual(47)
  })

  test("Orange and orange", () => {
    const resistorColor = new ResistorColor(["orange", "orange"])
    expect(resistorColor.value()).toEqual(33)
  })

  test("Ignore additional colors", () => {
    const resistorColor = new ResistorColor(["green", "brown", "orange"])
    expect(resistorColor.value()).toEqual(51)
  })
  test("Throws error when not enough colors", () => {
    expect(() => new ResistorColor(["green"])).toThrowError(
      "At least two colors need to be present"
    )
  })
})

File: dart/resistor-color-duo/test/resistor_color_duo_test.dart
import 'package:resistor_color_duo/resistor_color_duo.dart';
import 'package:test/test.dart';

void main() {
  final resistorColorDuo = ResistorColorDuo();

  group('ResistorColorDuo', () {
    test('Brown and black', () {
      final result = resistorColorDuo.value(<String>['brown', 'black']);
      expect(result, equals(10));
    }, skip: false);

    test('Blue and grey', () {
      final result = resistorColorDuo.value(<String>['blue', 'grey']);
      expect(result, equals(68));
    }, skip: false);

    test('Yellow and violet', () {
      final result = resistorColorDuo.value(<String>['yellow', 'violet']);
      expect(result, equals(47));
    }, skip: false);

    test('White and red', () {
      final result = resistorColorDuo.value(<String>['white', 'red']);
      expect(result, equals(92));
    }, skip: false);

    test('Orange and orange', () {
      final result = resistorColorDuo.value(<String>['orange', 'orange']);
      expect(result, equals(33));
    }, skip: false);

    test('Ignore additional colors', () {
      final result = resistorColorDuo.value(<String>['green', 'brown', 'orange']);
      expect(result, equals(51));
    }, skip: false);

    test('Black and brown, one-digit', () {
      final result = resistorColorDuo.value(<String>['black', 'brown']);
      expect(result, equals(1));
    }, skip: false);
  });
}
